lib/plugins.js

```diff
@@ -0,0 +1,68 @@
+const logger = require('./logger').plugins;
+const eventBus = require('./eventBus');
+const engineVersion = 'v1.0';
+module.exports.load = function ({config}) {
+  config = config || require('./config');
+  let pluginsSettings = config.systemConfig.plugins || {};
+
+  let pluginConfig = config.systemConfig.plugins || {};
+
+  let loadedPlugins = [];
+  logger.debug('Loading plugins. Plugin engine version:', engineVersion);
+  for (let pluginName in pluginConfig) {
+    try {
+      logger.debug('Loading plugin', pluginName);
+      let settings = pluginsSettings[pluginName];
+      let requireName = (settings && settings.package) || pluginName;
+      let plugin = require(requireName);
+      if (plugin.version !== engineVersion) { // TODO: versioning of plugins
+        logger.warn(plugin.version, 'is different from engine version:', engineVersion, 'trying to load');
+      }
+      let manifest = new PluginManifest({settings, config});
+      plugin.init(manifest);
+      loadedPlugins.push(manifest);
+      logger.info('Loaded plugin', pluginName);
+    } catch (err) {
+      logger.error('Failed to load plugin ' + pluginName, err);
+    }
+  }
+
+  // Note: All logic to handle different plugin version should be here
+  // Note: Rest of EG code must use only one standard interface
+  return {
+    policies: extract(loadedPlugins, 'policies'),
+    conditions: extract(loadedPlugins, 'conditions'),
+    gatewayExtensions: extract(loadedPlugins, 'gatewayExtensions'),
+    adminExtensions: extract(loadedPlugins, 'adminExtensions'),
+    cliExtensions: extract(loadedPlugins, 'cliExtensions')
+  };
+};
+
+class PluginManifest {
```

Comment:
I think the concept of "manifest" is a little strange in this context.  I think of a manifest as a file living outside the code that contains metadata about the contents of the project, usually written in a config format like XML, JSON, or INI.  I get that it's serving a similar purpose, but I look at this in code and think, "Gee, is this telling me I can actually put this manifest in a separate config file?"  This might be a bias based on my past experiences.  Not sure.  Also, feels a tad more weird when we start assigning event handlers.  Just thought I'd toss this out there.  Would it make more sense to just call this... `Plugin`?

Reply:
I agree that it looks weird to common approach like that  
https://developer.chrome.com/extensions/manifest

the thing is that what it passes is not a plugin, because plugin is entire package itself. 
It is something like container/ plugin manager / context. 

I cannot name it pluginOptions because it will be confused with parameters and settings